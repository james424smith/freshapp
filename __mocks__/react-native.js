import * as ReactNative from "react-native";

export * from "react-native";
export const Vibration = { vibrate: jest.fn() };
export const Adnimated = { spring: jest.fn() };

export default Object.setPrototypeOf(
  {
    Vibration,
    Adnimated,
  },
  ReactNative
);
