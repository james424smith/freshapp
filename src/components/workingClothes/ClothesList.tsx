import React from "react";
import { View } from "react-native";
import Text from "../StyledText";
import styles from "../../styles/more_screens/workingClothesStyles";
import { ClotheData } from "../../interfaces/workingClothesTypes";
import valueOrDefault from "../../common/valueOrDefault";
import { useTheme, Theme } from "@react-navigation/native";
type Props = { clothesTable: ClotheData[] };

const renderWorkingClothesItem = (
  theme: Theme,
  clotheItem: ClotheData,
  key: number
) => {
  return (
    <View
      key={key}
      style={{
        ...styles.clotheItemView,
        backgroundColor: theme.colors.primary,
      }}
    >
      <Text style={{ ...styles.dateValueSmall, color: theme.colors.text }}>
        {valueOrDefault(clotheItem.issueDate, "")}
      </Text>
      <View>
        <View style={styles.itemContainer}>
          <Text
            style={{ ...styles.detailsValueLarge, color: theme.colors.text }}
          >
            {valueOrDefault(clotheItem.item, "")}
          </Text>
        </View>
        <View style={styles.moreDetailsContainer}>
          <View style={styles.clotheLabels}>
            <Text
              style={{ ...styles.labelsValueBold, color: theme.colors.text }}
            >
              {"Qty:"}
            </Text>
            <Text
              style={{ ...styles.labelsValueBold, color: theme.colors.text }}
            >
              {"Agent:"}
            </Text>
            <Text
              style={{ ...styles.labelsValueBold, color: theme.colors.text }}
            >
              {"Vessel:"}
            </Text>
          </View>
          <View style={styles.moreDetailsValues}>
            <Text
              style={{ ...styles.detailsValueLarge, color: theme.colors.text }}
            >
              {valueOrDefault(clotheItem.quantity, "")}
            </Text>
            <Text
              style={{ ...styles.detailsValueLarge, color: theme.colors.text }}
            >
              {valueOrDefault(clotheItem.issuingAgentName, "")}
            </Text>
            <Text
              style={{ ...styles.detailsValueLarge, color: theme.colors.text }}
            >
              {valueOrDefault(clotheItem.vesselName, "")}
            </Text>
          </View>
        </View>
      </View>
      <View style={styles.lineStyle} />
    </View>
  );
};

const ClotheItemsList = (props: Props) => {
  const { clothesTable } = props;
  const theme = useTheme();
  return (
    <View
      style={{
        ...styles.DataContainerRoot,
        backgroundColor: theme.colors.primary,
      }}
    >
      <View style={styles.clothesContainer}>
        <Text style={{ ...styles.clothesTitle, color: theme.colors.text }}>
          Clothes Allocation
        </Text>
        <View style={styles.clotheDataList}>
          {clothesTable.map((clotheItem: ClotheData, i: number) =>
            renderWorkingClothesItem(theme, clotheItem, i)
          )}
        </View>
      </View>
    </View>
  );
};

export default ClotheItemsList;
