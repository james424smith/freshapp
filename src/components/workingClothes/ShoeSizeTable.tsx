import React from "react";
import { View } from "react-native";
import Text from "../StyledText";
import styles from "../../styles/more_screens/workingClothesStyles";
import { Table, Row, Rows } from "react-native-table-component";
import valueOrDefault from "../../common/valueOrDefault";
import { useTheme } from "@react-navigation/native";

type Props = { headings: string[]; sizeData: (number | undefined)[][] };

const ShoeSizeTable = (props: Props) => {
  const theme = useTheme();
  const { headings, sizeData } = props;
  return (
    <View style={styles.DataContainerRoot}>
      <View
        style={{
          ...styles.shoesContainer,
          backgroundColor: theme.colors.fromBlackToBlue,
        }}
      >
        <View style={styles.tableMarginsStyle}>
          <Text style={styles.shoesTitle}>Shoe Size</Text>
          <View style={styles.tableContainer}>
            <Table
              borderStyle={styles.tableBorderStyle}
              style={{
                ...styles.tableFillStyle,
                backgroundColor: theme.colors.background,
              }}
            >
              <Row
                data={headings}
                style={styles.headingStyle}
                textStyle={{
                  ...styles.dataHeading,
                  color: theme.colors.darkColorOfTable,
                }}
              />
              <Rows
                data={valueOrDefault(sizeData, []) as (number | undefined)[][]}
                textStyle={{
                  ...styles.dataRow,
                  color: theme.colors.darkColorOfTable,
                }}
              />
            </Table>
          </View>
        </View>
      </View>
    </View>
  );
};

export default ShoeSizeTable;
