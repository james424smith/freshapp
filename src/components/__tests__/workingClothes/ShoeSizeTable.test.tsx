import React from "react";
import ShoeSizeTable from "../../workingClothes/ShoeSizeTable";
import { render, cleanup } from "@testing-library/react-native";
import workingClothes from "../../../../__mocks__/fake-data/workingClothes.json";

describe("AppColorScheme snapshot test", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  it("should render correct ui and press privacy-policy-button", () => {
    const { toJSON } = render(
      <ShoeSizeTable
        headings={["PHL", "US", "EU", "UK"]}
        sizeData={[
          [
            workingClothes.shoeSizePhl,
            workingClothes.shoeSizeUs,
            workingClothes.shoeSizeEu,
            workingClothes.shoeSizeUk,
          ],
        ]}
      />
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
