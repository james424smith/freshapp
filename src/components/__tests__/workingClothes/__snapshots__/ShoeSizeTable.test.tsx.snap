// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppColorScheme snapshot test should render correct ui and press privacy-policy-button 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f6f6f6",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": undefined,
      }
    }
  >
    <View
      style={
        Object {
          "margin": 45,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2A2E43",
              "fontFamily": "Roboto-Regular",
            },
            Object {
              "color": "#FFFFFF",
              "fontSize": 33.5,
              "fontWeight": "bold",
              "textAlign": "left",
            },
          ]
        }
      >
        Shoe Size
      </Text>
      <View
        style={
          Object {
            "marginBottom": 45,
            "marginTop": 45,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgb(242, 242, 242)",
                "borderBottomEndRadius": 0.5,
                "borderBottomRightRadius": 0.5,
                "borderRadius": 2,
              },
              Object {
                "borderBottomWidth": 0.5,
                "borderColor": "#005AA5",
                "borderLeftWidth": 0.5,
              },
            ]
          }
        >
          <View
            style={
              Array [
                undefined,
                0,
                Object {
                  "flexDirection": "row",
                  "overflow": "hidden",
                },
                Object {
                  "height": 80,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "borderColor": "#005AA5",
                    "borderRightWidth": 0.5,
                    "borderTopWidth": 0.5,
                  },
                  Object {
                    "justifyContent": "center",
                  },
                  undefined,
                  undefined,
                  undefined,
                  Object {
                    "flex": 1,
                  },
                  undefined,
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 26.5,
                      "textAlign": "center",
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ]
                }
              >
                PHL
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "borderColor": "#005AA5",
                    "borderRightWidth": 0.5,
                    "borderTopWidth": 0.5,
                  },
                  Object {
                    "justifyContent": "center",
                  },
                  undefined,
                  undefined,
                  undefined,
                  Object {
                    "flex": 1,
                  },
                  undefined,
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 26.5,
                      "textAlign": "center",
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ]
                }
              >
                US
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "borderColor": "#005AA5",
                    "borderRightWidth": 0.5,
                    "borderTopWidth": 0.5,
                  },
                  Object {
                    "justifyContent": "center",
                  },
                  undefined,
                  undefined,
                  undefined,
                  Object {
                    "flex": 1,
                  },
                  undefined,
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 26.5,
                      "textAlign": "center",
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ]
                }
              >
                EU
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "borderColor": "#005AA5",
                    "borderRightWidth": 0.5,
                    "borderTopWidth": 0.5,
                  },
                  Object {
                    "justifyContent": "center",
                  },
                  undefined,
                  undefined,
                  undefined,
                  Object {
                    "flex": 1,
                  },
                  undefined,
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 26.5,
                      "textAlign": "center",
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ]
                }
              >
                UK
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                0,
                0,
              ]
            }
          >
            <View
              style={
                Array [
                  undefined,
                  0,
                  Object {
                    "flexDirection": "row",
                    "overflow": "hidden",
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "borderColor": "#005AA5",
                      "borderRightWidth": 0.5,
                      "borderTopWidth": 0.5,
                    },
                    Object {
                      "justifyContent": "center",
                    },
                    undefined,
                    undefined,
                    undefined,
                    Object {
                      "flex": 1,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 25.5,
                        "fontWeight": "bold",
                        "height": 57.5,
                        "margin": 13.5,
                        "textAlign": "center",
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                    ]
                  }
                >
                  8.5
                </Text>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "borderColor": "#005AA5",
                      "borderRightWidth": 0.5,
                      "borderTopWidth": 0.5,
                    },
                    Object {
                      "justifyContent": "center",
                    },
                    undefined,
                    undefined,
                    undefined,
                    Object {
                      "flex": 1,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 25.5,
                        "fontWeight": "bold",
                        "height": 57.5,
                        "margin": 13.5,
                        "textAlign": "center",
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                    ]
                  }
                >
                  42
                </Text>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "borderColor": "#005AA5",
                      "borderRightWidth": 0.5,
                      "borderTopWidth": 0.5,
                    },
                    Object {
                      "justifyContent": "center",
                    },
                    undefined,
                    undefined,
                    undefined,
                    Object {
                      "flex": 1,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 25.5,
                        "fontWeight": "bold",
                        "height": 57.5,
                        "margin": 13.5,
                        "textAlign": "center",
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                    ]
                  }
                >
                  8
                </Text>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "borderColor": "#005AA5",
                      "borderRightWidth": 0.5,
                      "borderTopWidth": 0.5,
                    },
                    Object {
                      "justifyContent": "center",
                    },
                    undefined,
                    undefined,
                    undefined,
                    Object {
                      "flex": 1,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 25.5,
                        "fontWeight": "bold",
                        "height": 57.5,
                        "margin": 13.5,
                        "textAlign": "center",
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                    ]
                  }
                >
                  8
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
